### PR TITLE
Code refactoring And New Features

### PR DESCRIPTION
**Pull request by PrajwalM2212**
### Description
This pull request changes activity to AppCompatActivity and removes android:configChanges in the manifest file . It also removes Null Pointer Exceptions present in the project. This pull request adds a new feature to the existing project . It creates a settings screen and a notification service. The user can switch on or off a alarm to remind them to use the app at a chosen time. The app sends notifications at the selected time every day reminding the user to use the app.

Fixes # [ISSUE]

### Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
This only serves refactor purpose and does not alter functionality of the existing code. The app runs as usual. The new feature appears as images below

settings
notification

### Checklist:
**Delete irrelevant options.**

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Rimjhim28** [comment]
@PrajwalM2212 Your PR goes against the general contributing guidelines of the community. 
Please go through the guidelines to avoid trouble in future.
You can view them [here](https://github.com/systers/powerup-android/wiki/How-to-Contribute)

File: PowerUp/app/src/main/java/powerup/systers/com/SettingsActivity.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package powerup.systers.com;
+
+import android.support.v4.app.FragmentManager;
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/SettingsActivity.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/SettingsActivity.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import android.support.v4.app.FragmentTransaction;
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/SettingsActivity.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/SettingsFragment.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import powerup.systers.com.alarm.AlarmReceiver;
+
+/**
+ * Created by prajwalm on 31/01/18.
+ */
+
+public class SettingsFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {
+
+
+    public SettingsFragment() {
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/SettingsFragment.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/StartActivity.java [hunk 4]
```diff
@@ -17,25 +17,33 @@
...
 import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
 import android.view.Gravity;
 import android.view.View;
 import android.view.WindowManager;
 import android.widget.Button;
 import android.widget.Toast;
 
-public class StartActivity extends Activity {
+import powerup.systers.com.alarm.AlertService;
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/StartActivity.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/StartActivity.java [hunk 5]
```diff
@@ -17,25 +17,33 @@
...
+
+public class StartActivity extends AppCompatActivity {
 
     private SharedPreferences preferences;
     private boolean hasPreviouslyStarted;
     private Button startButton;
     private Button newUserButton;
     private Button aboutButton;
+    private Button settingsButton;
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/StartActivity.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/alarm/AlarmManagerProvider.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Created by prajwalm on 31/01/18.
+ */
+
+public class AlarmManagerProvider {
+
+    private static AlarmManager sAlarmManager;
+
+    static synchronized AlarmManager getAlarmManager(Context context) {
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/alarm/AlarmManagerProvider.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/alarm/AlertService.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
+package powerup.systers.com.alarm;
+
+import android.app.IntentService;
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/alarm/AlertService.java]: I have fixed it

File: PowerUp/app/src/main/java/powerup/systers/com/SettingsActivity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply PowerUp/app/src/main/java/powerup/systers/com/SettingsActivity.java]: I have fixed it

**Comment by MeepyMay** [comment]
New feature wasn't previously discussed, so closing this PR for now.

**Pull request closed without merging**